### PR TITLE
Add tests for widget functionality

### PR DESCRIPTION
**Pull request by psobolewskiPhD**
Adding tests, to address https://github.com/jdeschamps/napari-sklearn-decomposition/issues/6
Hopefully with CI fixed, they will run!
Currently it's just the most basic tests:
1. that sample data is loaded correctly
2. that a widget is created
(Note: I commented out the example tests because they fail, but for now leaving them as a reference.)

### CODE REVIEW

**Comment by psobolewskiPhD** [comment]
I fleshed out the tests a bit.
The first test just checks the main widget, which is called by `decomposition()`. It makes sure that it works and that the sub-widget is also properly initialized with default values.
In the 2nd test, the 3 implemented methods are tested: that they make an output image with the correct name and shape.

File: src/napari_sklearn_decomposition/_tests/test_widget.py [hunk 1]
```diff
@@ -1,33 +1,94 @@
...
+    viewer.add_image(np.random.random((100, 50, 50)))
+    # Launch the widget for the decomposition method
+    widget = method()
+    viewer.window.add_dock_widget(widget)
+    # Run the widget with default settings
+    widget()
+    # Check a new image was added
+    assert len(viewer.layers) == 2
+    # Check that the name & shape of the new layer is correct
+    assert viewer.layers[-1].name == f"{method.__name__} result"
```
Comment (zoccoler) [inline src/napari_sklearn_decomposition/_tests/test_widget.py:59]: Just a reminder to update this line before merging #9

**Review by zoccoler — APPROVED** [review]
Great, we finally have passing functional tests! 🎉 
Thanks Peter @psobolewskiPhD!

File: src/napari_sklearn_decomposition/_tests/test_widget.py (hunk 1 above)
Comment (psobolewskiPhD) [reply src/napari_sklearn_decomposition/_tests/test_widget.py:59]: Thanks! Probably thinking this gets merged first, then I fix it in the other PR.

**Pull request merged**